cvecache panics on eviction

Description
=========

When using `cpe2cve` with a positive `-cache_size` (0 and negative work fine) a panic occurs:

```
panic: attempted to evict non-existent record

goroutine 12 [running]:
github.com/facebookincubator/nvdtools/cvefeed.(*Cache).evict(0xc0046dedc0, 0x0)
	/home/s/go/src/github.com/facebookincubator/nvdtools/cvefeed/cvecache.go:238 +0x12a
github.com/facebookincubator/nvdtools/cvefeed.(*Cache).Get(0xc0046dedc0, 0xc00000e5a0, 0x1, 0x1, 0x1, 0xc0059d4110, 0x10)
	/home/s/go/src/github.com/facebookincubator/nvdtools/cvefeed/cvecache.go:171 +0x425
main.processAll(0xc0059d2600, 0xc0059d2660, 0xc0046c4d20, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/s/nvdtools/cmd/cpe2cve/cpe2cve.go:73 +0xd43
main.processInput.func1(0xc0059d2600, 0xc0059d2660, 0xc0046c4d20, 0xc00008c0f0, 0xc0059d40d8, 0xc0059d40e0)
	/home/s/nvdtools/cmd/cpe2cve/cpe2cve.go:139 +0xcc
created by main.processInput
	/home/s/nvdtools/cmd/cpe2cve/cpe2cve.go:138 +0x309
```

To reproduce
==========

`echo "cpe:/a:foo:bar:*" | ./cpe2cve -cpe 1 -e 1 -cve 1 -cache_size 1 ./nvdcve-1.1-2002.json.gz `  
This bug doesn't happen when the cache size is large enough (432 for the 2002 feed).

--- a/cvefeed/cvecache.go
+++ b/cvefeed/cvecache.go
@@ -166,10 +166,10 @@ func (c *Cache) Get(cpes []*wfn.Attributes) []MatchResult {
 	cves.res = c.match(cpes)
 	cves.updateResSize(key)
 	c.mu.Lock()
-	c.size += cves.size
-	if c.MaxSize != 0 && c.size > c.MaxSize {
-		c.evict(int64(cacheEvictPercentage * float64(c.MaxSize)))
+	if c.MaxSize != 0 && c.size+cves.size > c.MaxSize {
+		c.evict(int64(cacheEvictPercentage*float64(c.MaxSize)) + cves.size)
 	}
+	c.size += cves.size
 	cves.evictionIndex = c.evictionQ.push(key)
 	c.mu.Unlock()
 	close(cves.ready)
@@ -231,7 +231,7 @@ func (c *Cache) matchDict(cpes []*wfn.Attributes, dict Dictionary) (results []Ma
 // evict the least recently used records untile nbytes of capacity is achieved or no more records left.
 // It is not concurrency-safe, c.mu should be locked before calling it.
 func (c *Cache) evict(nbytes int64) {
-	for c.size+nbytes > c.MaxSize {
+	for c.size > 0 && c.size+nbytes > c.MaxSize {
 		key := c.evictionQ.pop()
 		cd, ok := c.data[key]
 		if !ok { // should not happen

